homeassistant/components/rachio/switch.py

```diff
@@ -278,12 +278,12 @@ def _handle_update(self, *args, **kwargs) -> None:
         elif args[0][KEY_SUBTYPE] in [SUBTYPE_ZONE_STOPPED, SUBTYPE_ZONE_COMPLETED]:
             self._state = False
 
-        self.schedule_update_ha_state()
+        self.async_write_ha_state()
 
     async def async_added_to_hass(self):
         """Subscribe to updates."""
         self._undo_dispatcher = async_dispatcher_connect(
```

Comment:
We can use `self.async_on_remove` to unsubscribe the dispatch on entity removal instead. Please remove `self._undo_dispatcher`.